### DISCUSSION TITLE
Load settings from a local file

### DISCUSSION THREAD

**Discussion by Shamaoke** [brookhong/Surfingkeys#1892] [Q&A] [answered]
Hi.

How can I load settings from a local file? I tried to set `file:///home/user/.config/surfingkeys/settings.js` as a value of the “Load settings from” field in the extension settings page, but when I clicked the “Save” button, I got the “Failed to load settings from file:///home/.../settings.js” error message.

Thanks.

Arch Linux
Firefox 106
Surfingkeys 0.9.74

**Comment by yusufaktepe** [comment]
Firefox doesn't support loading settings from a local file. That's only available for Chromium-based browsers.

**Comment by Shamaoke** [comment, marked as answer]
I solved the problem by running a local web server to serve the Surfingkeys configuration file.

Here's how I've done this.

**1.** Place the `settings.js` Surfingkeys configuration file in the `/home/user/.config/surfingkeys/` directory;

**2.** Install the Ruby programming language: `pacman -S ruby`;

**3.** Install the WEBrick server: `gem install webrick`;

**4.** Create the `surfingkeys.service` user systemd unit in the `/home/user/.config/systemd/user/` directory with the following content;

````
[Unit]
Description=Run the WEBrick server to serve the Surfingkeys configuration file

[Service]
ExecStart=/usr/bin/ruby -r un -e httpd /home/user/.config/surfingkeys -b 127.0.0.1 -p 8888

[Install]
WantedBy=default.target
````

**5.** Enable the unit: `systemd --user enable surfingkeys.service`. The unit will be started automatically at every login. It will run the web server which will serve the surfingkeys configuration file;

**6.** Input the Surfingkeys configuration file URL `http://127.0.0.1:8888/settings.js` into the “Load settings from” field on the Surfingkeys configuration page in Firefox  (about:addons → Surfingkeys → Preferences) and press the “Save” button. The file will be read by the extension and the configuration preferences will be enabled.

**Reply by xiaobing-huang** [reply]
[dufs](https://github.com/sigoden/dufs) can also be used to server surfingkeys configuration file.
